commcare-cloud/commcare_cloud/commands/ansible/ansible_playbook.py

```diff
@@ -310,19 +318,133 @@ def make_parser(self):
                 "Example Usage: --only=riak,stanchion"
             )
         )
+        self.parser.add_argument(
+            '--worker_name',
+            help=(
+                "Celery worker name as appearing in the status list. "
+                "This can also be a comma-separated list of workers"
+                "For ex: Run 'commcare-cloud staging celery status' to get name of workers"
+            )
+        )
 
     def get_inventory_group_for_service(self, service, service_group):
         return self.INVENTORY_GROUP_FOR_SERVICE.get(service, service_group)
 
+    @staticmethod
+    def _get_celery_processes(env):
+        environment = get_environment(env)
+        app_processes_config = environment.translated_app_processes_config
+        return app_processes_config.celery_processes
+
+    def get_celery_config(self, args):
+        environment = get_environment(args.environment)
+        project_name = environment.fab_settings_config.project
+        """
+        This picks the logic from set_celery_supervisorconf to produce the full worker names
+        as declared in the template supervisor_celery_worker.conf
+        {{ project }}-{{ environment }}-celery_{{ celery_params.comma_separated_queue_names }}_{{ celery_params.worker_num }}
```

Comment:
Careful, this `environment` variable here is not the same as `args.environment` (sorry!). In fab, it gets populated from the `environment` property of the AppProcessesConfig. As you can see they are very often the same:
```
$ grep environment environments/*/app-processes.yml
environments/enikshay/app-processes.yml:environment: enikshay
environments/icds-new/app-processes.yml:environment: icds
environments/icds/app-processes.yml:environment: icds
environments/l10k/app-processes.yml:environment: l10k
environments/pna/app-processes.yml:environment: pna
environments/production/app-processes.yml:environment: production
environments/softlayer/app-processes.yml:environment: softlayer
environments/staging/app-processes.yml:environment: staging
environments/swiss/app-processes.yml:environment: swiss
```
but different for `icds-new`. `environment.translated_app_processes_config.environment` should get you what you're looking for.

Reply:
oh ho.. this is a bit confusing..
i am going to take icds-new example

so `icds-new` is the environment
where `icds` should be called the `environment_environment` or `environment_name` or `commcare_environment`?

also `env` is not always an `Environment` object. This is similar to what we have in HQ were domain in code is at times the name of the domain or the `Domain` object. So we try to use project for `Domain` object but its not reliable still. 
Can we avoid that here by using `env_name` for the `name` like `icds-new`, and `env_obj` for `Environment` class object and `env` for the `fab.api.env` module.